sdk/storage_datalake/src/shared_key_authorization_policy.rs

```diff
@@ -0,0 +1,244 @@
+use azure_core::{Context, Policy, PolicyResult, Request, Response};
+use http::{HeaderMap, HeaderValue, Method};
+use ring::hmac;
+use std::sync::Arc;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SharedKeyAuthorizationPolicy {
+    base_url: String,
+    shared_key: String,
+    storage_account_name: String,
+}
+
+impl SharedKeyAuthorizationPolicy {
+    #[allow(unused)]
+    pub(crate) fn new(base_url: String, shared_key: String, storage_account_name: String) -> Self {
+        Self {
+            base_url,
+            shared_key,
+            storage_account_name,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Policy for SharedKeyAuthorizationPolicy {
+    async fn send(
+        &self,
+        ctx: &Context,
+        request: &mut Request,
+        next: &[Arc<dyn Policy>],
+    ) -> PolicyResult<Response> {
+        if next.is_empty() {
+            return Err(Box::new(azure_core::PipelineError::InvalidTailPolicy(
+                "Authorization policies cannot be the last policy of a pipeline".to_owned(),
+            )));
+        }
+
+        let headers_mut = request.headers_mut();
+        headers_mut.append(
+            azure_core::headers::MS_DATE,
+            HeaderValue::from_str(
+                format!("{}", chrono::Utc::now().format("%a, %d %h %Y %T GMT")).as_str(),
```

Comment:
Does this work?
```suggestion
                &chrono::Utc::now().format("%a, %d %h %Y %T GMT").to_string()
```

Reply:
For this PR, I'd like to change as little as possible in this code, since it's directly copied from the pre-pipeline shared key auth code in storage_account_client.rs. If there are any issues there, I'd like to address them in a follow-up PR.

Reply:
Yes, fixed in follow-up PR.